clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -1290,31 +1290,38 @@ inline void import_external_semaphore(external_sem_wrapper **extSem,
 /// parameters.
 /// \param [in] numExtSems Number of external semaphores to signal.
 /// \param [in] q_ptr The queue used to signal the external semaphore resource.
-inline void signal_external_semaphore(external_sem_wrapper **extSem,
-                                      external_sem_params *semSignalParams,
-                                      unsigned int numExtSems,
-                                      queue_ptr q_ptr = &get_default_queue()) {
+inline void signal_external_semaphores(external_sem_wrapper **extSem,
+                                       external_sem_params *semSignalParams,
+                                       unsigned int numExtSems,
+                                       queue_ptr q_ptr = &get_default_queue()) {
   for (int i = 0; i < numExtSems; i++) {
     auto extSemType = extSem[i]->get_handle_type();
+    switch (extSemType) {
 #ifdef _WIN32
-    if (extSemType == sycl::ext::oneapi::experimental::
-                          external_semaphore_handle_type::win32_nt_dx12_fence) {
-      q_ptr->ext_oneapi_signal_external_semaphore(
+    case sycl::ext::oneapi::experimental::external_semaphore_handle_type::
```

Comment:
What is the difference of cases? And can be merge WIN32 together?

Reply:
Okay, I'll refactor the switch case